### PR TITLE
Set connection attributes in SingleStoreIO

### LINKED ISSUE #27488: [Task]: Set connection attributes in SingleStoreIO
### What needs to happen?

When a connection to the SingleStore is established, it should pass connection attributes. Setting connection attributes allows SingleStore to get more visibility on what connectors are used by customers and prioritize work on them.

### Issue Priority

Priority: 3 (nice-to-have improvement)

### Issue Components

- [ ] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by AdalbertMemSQL**
Setting connection attributes allows SingleStore to get more visibility on what connectors are used by customers and prioritize work on them.
fixes #27488 
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/java/io/singlestore/src/test/java/org/apache/beam/sdk/io/singlestore/SingleStoreIOConnectionAttributesIT.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      while (rs.next()) {
+        String attribute = rs.getString(3);
+        String value = rs.getString(4);
+        if (attributes.containsKey(attribute)) {
+          assertEquals(attributes.get(attribute), value);
+          attributes.remove(attribute);
+        }
+      }
+    } catch (Exception e) {
+      File file = new File("/home/amakarovych-ua/Test/log");
```
Comment (pmishchenko-ua) [inline sdks/java/io/singlestore/src/test/java/org/apache/beam/sdk/io/singlestore/SingleStoreIOConnectionAttributesIT.java:91]: should this path be more generic?

**Comment by AdalbertMemSQL** [comment]
assign to next reviewer @johnjcasey

**Comment by AdalbertMemSQL** [comment]
R: @johnjcasey

**Review by pmishchenko-ua — APPROVED** [review]

**Review by johnjcasey — APPROVED** [review]

**Comment by damccorm** [comment]
@johnjcasey @AdalbertMemSQL what are next steps for this PR?

**Review by johnjcasey — APPROVED** [review]

**Pull request merged**